docs/discussion/TermComments.md

```diff
@@ -0,0 +1,25 @@
+# Term Comments
```

Comment:
Turn more into an explanation type document and provide explanation as well to how to "think" of comments.